### DISCUSSION TITLE
Union like in c

### DISCUSSION THREAD

**Discussion by tobias-mierzwa** [pydantic/pydantic#4527] [Question] [answered]
Hey together,

is there a way to use Unions like in C? Right now, my code is like in the following snippet. My JSON has the possible to have either a coding List with CodingItem2 items or text string. What i pref is a way like in c where It's either `union {coding: List[CodingItem2], text: str}` instead of two optional. Is there a way to make this possible, or a better way then I have right now?

```python
class CodingItem2(BaseModel):
    system: str
    code: str


class Form(BaseModel):
    coding: Optional[List[CodingItem2]] = None
    text: Optional[str]
```

**Comment by rafalkrupinski** [comment, marked as answer]
pydantic works fine with typing.Union

```python
class Form(BaseModel):
    coding_or_text: Union[List[CodingItem2], str, None] = None
```
